pkg/datapath/linux/config/config.go

```diff
@@ -1035,13 +1034,27 @@ func (h *HeaderfileWriter) writeStaticData(fw io.Writer, e datapath.EndpointConf
 func (h *HeaderfileWriter) WriteEndpointConfig(w io.Writer, e datapath.EndpointConfiguration) error {
 	fw := bufio.NewWriter(w)
 
+	nativeDevices, _ := tables.SelectedDevices(h.devices, h.db.ReadTxn())
+	deviceNames := tables.DeviceNames(nativeDevices)
+
+	// When WG & encrypt-node are on, a NodePort BPF to-be forwarded request
```

Comment:
Why is the headerfile writer the right place for this judicious hack? Could it live elsewhere?

Reply:
We need some of the static data to be written here (otherwise we'll fail with some errors like `symbol 'ETH_HLEN' not found`)

I think the truly clean solution would be to have a separate table to know which programs want to be loaded on which devices, but we're not there yet. I think this is better that the original hack, which was just adding it to the devices globally.

Reply:
@joamaki you wanted to check with @brb regarding this - lmk of the resolution